gnosis/safe/safe.py

```diff
@@ -519,16 +522,28 @@ def contract(self) -> Contract:
 
     @cached_property
     def domain_separator(self):
-        return fast_keccak(
-            encode_abi(
-                ["bytes32", "uint256", "address"],
-                [
-                    self.DOMAIN_TYPEHASH,
-                    self.ethereum_client.get_chain_id(),
-                    self.address,
-                ],
+        version = self.contract.functions.VERSION().call()
```

Comment:
What about calling `retrieve_version()` here?

I think we can make it a `cached_property` in a separate PR since it affects other methods.

Reply:
Done https://github.com/safe-global/safe-eth-py/pull/422/commits/f1ac04392efe6c31df0dc636a931ade86c92155e